lib/matplotlib/_mathtext.py

```diff
@@ -2004,6 +2006,10 @@ def __init__(self):
 
         p.float_literal <<= Regex(r"[-+]?([0-9]+\.?[0-9]*|\.[0-9]+)")
         p.int_literal   <<= Regex("[-+]?[0-9]+")
+        p.unit          <<= Regex("(pt|mm|cm|in|em|mu|"
```

Comment:
I think you can get rid of the grouping parentheses, which should be marginally more efficient.